### DISCUSSION TITLE
Change `data` property of `useSignMessage` to `signature`

### DISCUSSION THREAD

**Discussion by BoxedFruits** [wevm/wagmi#1057] [Ideas]
![image](https://user-images.githubusercontent.com/34636700/194778101-1efb1f04-73ea-4540-9d37-070e9c043753.png)

The `data` property of `useSignMessage` is the signature that is returned after signing a message. I think it would make more sense if it was renamed to `signature` to reflect what it actually is.

**Comment by BoxedFruits** [comment]
I'd love to work on this change if others agree

**Comment by tmm** [comment]
Thanks for the feedback! This is one of those cases in API design where we are trying to go for standardization instead of contextual naming. This way folks can expect a `data` key across all hooks instead of needing to guess what the data return key is named (e.g. `signature`, `balance`, etc.). We follow similar standardization across other properties as well. For example, `isLoading` instead of `isSigning`.

That said, you can always rename `data` when destructing and give it a specific name for your app's context:

``` 
const { data: signature } = useSignMessage(…)
```

**Reply by BoxedFruits** [reply]
Makes sense! Also TIL that you can rename things while destructuring 📝
